package/base-files/files/lib/functions/caldata.sh

```diff
@@ -64,25 +76,30 @@ caldata_from_file() {
 
 	[ -n "$target" ] || target=/lib/firmware/$FIRMWARE
 
-	# dd doesn't handle partial reads from special files: use cat
-	cat $source | dd of=$target iflag=skip_bytes bs=$count skip=$offset count=1 2>/dev/null || \
+	caldata_dd $source $offset $count $target || \
 		caldata_die "failed to extract calibration data from $source"
 }
 
 caldata_sysfsload_from_file() {
 	local source=$1
 	local offset=$(($2))
 	local count=$(($3))
+	local target="/sys/$DEVPATH/data"
+	local target_dir=$(dirname "$target")
```

Comment:
What about:

```
local target_dir="/sys/$DEVPATH"
local target="$target_dir/data"
```

Should save one subshell invocation

Reply:
Done, thank you.